mixes/containers/mix.js

```diff
@@ -58,7 +59,20 @@ module.exports = connect(
   (state, ownProps) => {
     const props = getMixProps(state)
     const currentMixId = ownProps.params.mixId
-    return { ...props, mixId: currentMixId, mix: props.mixes[currentMixId] }
+    const { router, route } = ownProps
+    const mix = props.mixes[currentMixId]
+
+    if (mix) {
+      router.setRouteLeaveHook(
+        route,
+        // TODO: do i need to back out all changes if confirm? how to do that cleanly - LOAD_MIX? ROLLBACK_MIX?
```

Comment:
yeah `ROLLBACK_MIX` makes sense, which could call `LOAD_MIX` (or we store the original clean mix somewhere in the store).